### PR TITLE
Implement exponential moving average - Closes #106

### LINKED ISSUE #106: Implement estimated moving average algorithm
### Description

Implement estimated moving average algorithm, so it can be used in the LIP-0016 implementation.

Consider this thread on StackOverflow:

 https://stackoverflow.com/questions/40057020/calculating-exponential-moving-average-ema-using-javascript

### Motivation

- The in-house EMA implementation is needed for the estimated fee service.

### Acceptance Criteria

- The EMA is calculated properly
- Unit tests are written
- Positive review from the research team

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #106 

### How was it solved?
Implement function to calculate EMA based on LIP-0016

### How was it tested?
Locally

### CODE REVIEW

File: services/core/shared/core.js [hunk 1]
```diff
@@ -329,6 +329,36 @@ const getBlocks = async params => {
 const setReadyStatus = status => { readyStatus = status; };
 const getReadyStatus = () => readyStatus;
 
+const calcExpDecay = (NUM_BLOCKS, DECAY_RATE) => {
+	const expDecay = (1 - Math.pow(1 - DECAY_RATE, 1 / NUM_BLOCKS)).toFixed(5);
+	return expDecay;
+};
```
Comment (sameersubudhi) [inline services/core/shared/core.js:335]: 1. `calcExpDecay` method can be defined as an inner function within `EMAcalc`. `calcExpDecay` is a one-liner and not used anywhere else.
2. Let's keep the variable names consistent with the ones defined in the config file for the `calcExpDecay` method. `NUM_BLOCKS` to `emaBatchSize` and `DECAY_RATE` to be `emaDecayRate`.

File: services/core/shared/core.js [hunk 2]
```diff
@@ -329,6 +329,36 @@ const getBlocks = async params => {
...
+
+const EMAcalc = async (feePerByte, prevFeeEstPerByte) => {
```
Comment (sameersubudhi) [inline services/core/shared/core.js:337]: `EMAcalc` can be a synchronous method.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Just a few minor remarks to look into:

File: services/core/shared/core.js (hunk 1 above)
Comment (nagdahimanshu) [reply services/core/shared/core.js]: changes has been done as requested

File: services/core/shared/core.js (hunk 2 above)
Comment (nagdahimanshu) [reply services/core/shared/core.js]: Now `EMAcalc` is a synchronous function.

**Review by sameersubudhi — APPROVED** [review]
Thank you @nagdahimanshu. Looks good now!

**Pull request merged**